### ISSUE TITLE
Should we be pinning default versions for the Docker images used by Watchdog?

### ISSUE THREAD

**Issue by jkuester** [medic/cht-watchdog#80]
Currently we the default version for all the Docker images used by Watchdog is just `latest`. This has some pros and cons:

Pros:
- Convenient from a development standpoint since we do not have to worry about continually bumping the versions every time one of the images is upgraded
- Could be considered convenient for a prod deployment for the same reason (although would probably not be recommended due to stability concerns).  

Cons:
- We do not have any historic record of _which versions_ of the images actually worked with a particular version of Watchdog.  (Probably could compile information by checking release dates for versions of all the images, but that would be time-consuming and error-prone)
- Any deployments that do not set a custom version for the images will just be running against `latest`. If one of the images pushes a breaking change that is not supported by Watchdog it could break the deployment.
- If a deployment does set a custom version for the images, it is not clear, when upgrading Watchdog, if/when they need to also update the associated image versions.

## Proposal

I am thinking we should just set the latest versions as the default values instead of using `latest`.  Perhaps we could add some CI checks that would remind us to bump the version (or could Dependabot be configured to just submit a PR??).  This way, when you deployed Watchdog, you would not have to worry about manually managing all the image versions. Instead, if you do not have any custom versions configured, then you will automatically get the new image versions when upgrading Watchdog.

@mrjones-plip @m5r Thoughts?

**Comment by mrjones-plip** [comment]
yes, this is a nice split between pinned versions but that still can easily be upgraded when the default changes.  We'll loose the benefits of always being on `latest`, but gain the ability to know which docker image versions where pinned to which Watchdog version.

Ship it!

**Comment by m5r** [comment]
+1 for more consistency/stability :rocket:

**Comment by jkuester** [comment]
:+1: Sounds good!  Really, I imagine it would not be too difficult to roll-our-own GitHub action that would just compare our current version of the Docker images with the latest from DockerHub. :thinking: Not a big deal to do it manually for now, though!
